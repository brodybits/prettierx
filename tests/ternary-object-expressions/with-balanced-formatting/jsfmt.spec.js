// [prettierx] test script notice:
// This test script runs for test files in parent directory,
// **not** on any files in *this* directory.

const dirpath = `${__dirname}/..`;

run_spec(dirpath, ["babel", "babel-flow", "flow", "typescript"], {
  // [prettierx] balanced ternary formatting option
  // to improve consistency with "Standard JS" in certain cases
  // (may lead to inconsistencies in some other cases):
  alignTernaryLines: false,
  // [prettierx] more options needed for consistency with "Standard JS":
  arrowParens: "avoid",
  trailingComma: "none",
});

run_spec(dirpath, ["babel", "babel-flow", "flow", "typescript"], {
  // variation from ../jsfmt.spec.js:
  tabWidth: 4,
  // [prettierx] balanced ternary formatting option:
  alignTernaryLines: false,
  // [prettierx] more options needed for consistency with "Standard JS":
  arrowParens: "avoid",
  trailingComma: "none",
});

run_spec(dirpath, ["babel", "babel-flow", "flow", "typescript"], {
  // variation from ../jsfmt.spec.js:
  useTabs: true,
  // [prettierx] balanced ternary formatting option:
  alignTernaryLines: false,
  // [prettierx] more options needed for consistency with "Standard JS":
  arrowParens: "avoid",
  trailingComma: "none",
});

run_spec(dirpath, ["babel", "babel-flow", "flow", "typescript"], {
  // variation from ../jsfmt.spec.js:
  useTabs: true,
  tabWidth: 4,
  // [prettierx] balanced ternary formatting option:
  alignTernaryLines: false,
  // [prettierx] more options needed for consistency with "Standard JS":
  arrowParens: "avoid",
  trailingComma: "none",
});
