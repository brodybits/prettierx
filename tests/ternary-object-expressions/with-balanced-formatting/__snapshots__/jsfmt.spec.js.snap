// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`crewdress.js 1`] = `
====================================options=====================================
alignTernaryLines: false
arrowParens: "avoid"
parsers: ["babel", "babel-flow", "flow", "typescript"]
printWidth: 80
trailingComma: "none"
                                                                                | printWidth
=====================================input======================================
let isSpace = false

const dress = isSpace ? {
      spaceSuit: 3,
      oxygenCylinders: 6
    } : {
      shirts: 3,
      paints: 3
    }

=====================================output=====================================
let isSpace = false;

const dress = isSpace
  ? {
    spaceSuit: 3,
    oxygenCylinders: 6
  }
  : {
    shirts: 3,
    paints: 3
  };

================================================================================
`;

exports[`crewdress.js 2`] = `
====================================options=====================================
alignTernaryLines: false
arrowParens: "avoid"
parsers: ["babel", "babel-flow", "flow", "typescript"]
printWidth: 80
tabWidth: 4
trailingComma: "none"
                                                                                | printWidth
=====================================input======================================
let isSpace = false

const dress = isSpace ? {
      spaceSuit: 3,
      oxygenCylinders: 6
    } : {
      shirts: 3,
      paints: 3
    }

=====================================output=====================================
let isSpace = false;

const dress = isSpace
    ? {
        spaceSuit: 3,
        oxygenCylinders: 6
    }
    : {
        shirts: 3,
        paints: 3
    };

================================================================================
`;

exports[`crewdress.js 3`] = `
====================================options=====================================
alignTernaryLines: false
arrowParens: "avoid"
parsers: ["babel", "babel-flow", "flow", "typescript"]
printWidth: 80
trailingComma: "none"
useTabs: true
                                                                                | printWidth
=====================================input======================================
let isSpace = false

const dress = isSpace ? {
      spaceSuit: 3,
      oxygenCylinders: 6
    } : {
      shirts: 3,
      paints: 3
    }

=====================================output=====================================
let isSpace = false;

const dress = isSpace
	? {
		spaceSuit: 3,
		oxygenCylinders: 6
	}
	: {
		shirts: 3,
		paints: 3
	};

================================================================================
`;

exports[`crewdress.js 4`] = `
====================================options=====================================
alignTernaryLines: false
arrowParens: "avoid"
parsers: ["babel", "babel-flow", "flow", "typescript"]
printWidth: 80
tabWidth: 4
trailingComma: "none"
useTabs: true
                                                                                | printWidth
=====================================input======================================
let isSpace = false

const dress = isSpace ? {
      spaceSuit: 3,
      oxygenCylinders: 6
    } : {
      shirts: 3,
      paints: 3
    }

=====================================output=====================================
let isSpace = false;

const dress = isSpace
	? {
		spaceSuit: 3,
		oxygenCylinders: 6
	}
	: {
		shirts: 3,
		paints: 3
	};

================================================================================
`;

exports[`nested-ternary-promises.js 1`] = `
====================================options=====================================
alignTernaryLines: false
arrowParens: "avoid"
parsers: ["babel", "babel-flow", "flow", "typescript"]
printWidth: 80
trailingComma: "none"
                                                                                | printWidth
=====================================input======================================
// check for consistent formatting as discussed in:
// - https://github.com/eslint/eslint/issues/13971
// - https://github.com/standard/standard/issues/1624
// - https://github.com/brodybits/prettierx/issues/41

function test1() {
  return condition1
  ? condition2
    ? Promise(1)
    : Promise(2)
  : condition3
    ? Promise(3)
    : Promise(4)
}

=====================================output=====================================
// check for consistent formatting as discussed in:
// - https://github.com/eslint/eslint/issues/13971
// - https://github.com/standard/standard/issues/1624
// - https://github.com/brodybits/prettierx/issues/41

function test1() {
  return condition1
    ? condition2
      ? Promise(1)
      : Promise(2)
    : condition3
      ? Promise(3)
      : Promise(4);
}

================================================================================
`;

exports[`nested-ternary-promises.js 2`] = `
====================================options=====================================
alignTernaryLines: false
arrowParens: "avoid"
parsers: ["babel", "babel-flow", "flow", "typescript"]
printWidth: 80
tabWidth: 4
trailingComma: "none"
                                                                                | printWidth
=====================================input======================================
// check for consistent formatting as discussed in:
// - https://github.com/eslint/eslint/issues/13971
// - https://github.com/standard/standard/issues/1624
// - https://github.com/brodybits/prettierx/issues/41

function test1() {
  return condition1
  ? condition2
    ? Promise(1)
    : Promise(2)
  : condition3
    ? Promise(3)
    : Promise(4)
}

=====================================output=====================================
// check for consistent formatting as discussed in:
// - https://github.com/eslint/eslint/issues/13971
// - https://github.com/standard/standard/issues/1624
// - https://github.com/brodybits/prettierx/issues/41

function test1() {
    return condition1
        ? condition2
            ? Promise(1)
            : Promise(2)
        : condition3
            ? Promise(3)
            : Promise(4);
}

================================================================================
`;

exports[`nested-ternary-promises.js 3`] = `
====================================options=====================================
alignTernaryLines: false
arrowParens: "avoid"
parsers: ["babel", "babel-flow", "flow", "typescript"]
printWidth: 80
trailingComma: "none"
useTabs: true
                                                                                | printWidth
=====================================input======================================
// check for consistent formatting as discussed in:
// - https://github.com/eslint/eslint/issues/13971
// - https://github.com/standard/standard/issues/1624
// - https://github.com/brodybits/prettierx/issues/41

function test1() {
  return condition1
  ? condition2
    ? Promise(1)
    : Promise(2)
  : condition3
    ? Promise(3)
    : Promise(4)
}

=====================================output=====================================
// check for consistent formatting as discussed in:
// - https://github.com/eslint/eslint/issues/13971
// - https://github.com/standard/standard/issues/1624
// - https://github.com/brodybits/prettierx/issues/41

function test1() {
	return condition1
		? condition2
			? Promise(1)
			: Promise(2)
		: condition3
			? Promise(3)
			: Promise(4);
}

================================================================================
`;

exports[`nested-ternary-promises.js 4`] = `
====================================options=====================================
alignTernaryLines: false
arrowParens: "avoid"
parsers: ["babel", "babel-flow", "flow", "typescript"]
printWidth: 80
tabWidth: 4
trailingComma: "none"
useTabs: true
                                                                                | printWidth
=====================================input======================================
// check for consistent formatting as discussed in:
// - https://github.com/eslint/eslint/issues/13971
// - https://github.com/standard/standard/issues/1624
// - https://github.com/brodybits/prettierx/issues/41

function test1() {
  return condition1
  ? condition2
    ? Promise(1)
    : Promise(2)
  : condition3
    ? Promise(3)
    : Promise(4)
}

=====================================output=====================================
// check for consistent formatting as discussed in:
// - https://github.com/eslint/eslint/issues/13971
// - https://github.com/standard/standard/issues/1624
// - https://github.com/brodybits/prettierx/issues/41

function test1() {
	return condition1
		? condition2
			? Promise(1)
			: Promise(2)
		: condition3
			? Promise(3)
			: Promise(4);
}

================================================================================
`;
